### PR TITLE
Script for explaining predictions

### PR DESCRIPTION
**Pull request by PrimozGodec**
#### Description of changes
Script with functionalities to explain predictions

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

Before it works following PRs must be merged:
https://github.com/slundberg/shap/pull/1135 and Shap release
https://github.com/biolab/orange3/pull/4602 and Orange release

### CODE REVIEW

File: orangecontrib/prototypes/explanation/explainer.py [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+    progress_callback: Callable,
+) -> Tuple[List[np.ndarray], np.ndarray, np.ndarray]:
+    """
+    Computes SHAP values for any learner with KernelExplainer.
+    """
+    # 1000 is a number that for normal data and model do not take so long
+    data_sample, sample_mask = _subsample_data(transformed_data, 1000)
+
+    try:
+        ref = kmeans(transformed_reference_data.X, k=10)
```
Comment (VesnaT) [inline orangecontrib/prototypes/explanation/explainer.py:164]: Does this work for data being smaller than 10?

File: orangecontrib/prototypes/explanation/explainer.py [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+if __name__ == "__main__":
+    from Orange.classification import RandomForestLearner
+
+    data_ = Table.from_file("heart_disease.tab")
+    learner = RandomForestLearner()
+    model_ = learner(data_)
+
+    shap_val, transformed_domain, mask, colors_ = get_shap_values_and_colors(
+        model_, data_
+    )
```
Comment (VesnaT) [inline orangecontrib/prototypes/explanation/explainer.py:568]: If I run the following code, it crashes:
```
    from Orange.classification import RandomForestLearner, LogisticRegressionLearner

    data_ = Table.from_file("heart_disease.tab")[:1]
    learner = LogisticRegressionLearner()
    model_ = learner(data_)

    shap_val, transformed_domain, mask, colors_ = get_shap_values_and_colors(
        model_, data_
    )
```

File: orangecontrib/prototypes/explanation/explainer.py (hunk 1 above)
Comment (PrimozGodec) [reply orangecontrib/prototypes/explanation/explainer.py:165]: It works. It raises a ValueError which is caught and then sample is used instead of k-means

Comment (PrimozGodec) [reply orangecontrib/prototypes/explanation/explainer.py:165]: I added a test for such case

File: orangecontrib/prototypes/explanation/explainer.py (hunk 2 above)
Comment (PrimozGodec) [reply orangecontrib/prototypes/explanation/explainer.py:573]: It is fixed now.

**Pull request merged**